packages/manager/modules/pci/src/projects/project/private-registry/private-registry.service.js

```diff
@@ -0,0 +1,82 @@
+import head from 'lodash/head';
+import assign from 'lodash/assign';
+
+export default class pciPrivateRegistryService {
+  /* @ngInject */
+  constructor(
+    OvhApiCloudProject,
+    OvhApiMe,
+  ) {
+    this.OvhApiPrivateRegistry = OvhApiCloudProject.ContainerRegistry().v6();
+    this.OvhApiPrivateRegistryUser = OvhApiCloudProject.ContainerRegistry().Users().v6();
+    this.User = OvhApiMe;
+  }
+
+  /**
+   * create a new registry
+   * @accepts registry name
+   * @returns new registry
+   * @memberof pciPrivateRegistryService
+   */
+  createRegistry(projectId, registry) {
+    return this.OvhApiPrivateRegistry.create({ serviceName: projectId }, registry).$promise;
+  }
+
+  deleteRegistry(projectId, registryID) {
+    return this.OvhApiPrivateRegistry.delete({
+      serviceName: projectId,
+      registryID,
+    }).$promise;
+  }
+
+  getRegistryList(projectId, clearCache = false) {
+    if (clearCache) {
+      this.OvhApiPrivateRegistry.resetCache();
+      this.OvhApiPrivateRegistry.resetQueryCache();
+    }
+    return this.OvhApiPrivateRegistry.query({
+      serviceName: projectId,
+    }).$promise;
+  }
+
+  getRegistry(serviceName, registryID) {
+    return this.OvhApiPrivateRegistry.get({
+      serviceName, registryID,
+    }).$promise;
+  }
+
+  updateRegistry(projectId, registryID, newRegistryName) {
+    return this.OvhApiPrivateRegistry.update({
+      serviceName: projectId,
+      registryID,
+    }, {
+      name: newRegistryName,
+    }).$promise;
+  }
+
+  generateCredentials(projectId, registryID) {
+    return this.OvhApiPrivateRegistryUser.create({
+      serviceName: projectId,
+      registryID,
+    }).$promise;
+  }
+
+  getContractInfo(contractId) {
```

Comment:
If we want to keep a good pattern, contract should be a different service, maybe with some shortcut in this one.

```javascript
	pciPrivateRegistryContract.get()
	pciPrivateRegistryContract.accept()
```

AND

I think Agreemeents is generic for all product. Agreements service should be transversal to our products, else each product is going to implement Agreements `get` / `accept` methods.

Reply:
reusing newProject service